src/app.js

```diff
@@ -151,15 +153,29 @@ class Messenger extends EventEmitter {
     this.app.get('/ping', (req, res) => {
       res.send('pong');
     });
+
+    this.port = config.get('port');
+    this.isListening = 0;
+    this.start();
   }
 
   start() {
```

Comment:
Thinking of the edge case where you actually do want manual control, I had it in my head that the structure looked like:
```javascript
start(port = config.get('port')): Promise {
  return new Promise((resolve, reject) => {
    if (blah blah blah) {
      this.app.listen(port, (err) => {
        if (err) { this.isListening = 0; reject(err) }
        this.isListening++; debug(); resolve();
       });
    }
  });
}
```

This adds three things:
* extra complexity (boooo)
* less state because `port` is an argument (yaaaay)
* a Promise the caller can [hold on](https://www.youtube.com/watch?v=uIbXvaE39wM) for one more day (yaaay)

----

After I wrote that, another idea came into my head: since this is event based, you can do also do a `this.emit('appStarted')` too. That solves the problem of the caller not knowing what's going on inside. I'm not sure which way is better, but I get the feeling it's the Promise. the `throw err` might also want to be a `this.emit('error', err)` too. I don't have enough experience with Node event based SDKs to know what's expected. I usually just play with all the options with sample implementations and see which feels right.